### ISSUE TITLE
Expected warnings with Python 3.12

### ISSUE THREAD

**Issue by vsoch** [flux-framework/flux-core#5674]
There might be more, but I'm building flux-core in a container with Python 3.12 and am running into some warnings! Here they are (and fixes):

```console
  GEN      _idset_clean.h
/opt/flux-core/src/bindings/python/_flux/./make_clean_header.py:68: SyntaxWarning: invalid escape sequence '\s'
  m = re.search('#include\s*"([^"]*)"', l)
/opt/flux-core/src/bindings/python/_flux/./make_clean_header.py:72: SyntaxWarning: invalid escape sequence '\s'
  if not re.match("#\s*include", l):
  GEN      _idset_preproc.h
  GEN      _idset.c
  CC       _idset.lo
  CCLD     _idset.la
  GEN      _rlist_clean.h
/opt/flux-core/src/bindings/python/_flux/./make_clean_header.py:68: SyntaxWarning: invalid escape sequence '\s'
  m = re.search('#include\s*"([^"]*)"', l)
/opt/flux-core/src/bindings/python/_flux/./make_clean_header.py:72: SyntaxWarning: invalid escape sequence '\s'
  if not re.match("#\s*include", l):
  GEN      _rlist_preproc.h
  GEN      _rlist.c
  CC       _rlist.lo
  CCLD     _rlist.la
  GEN      _security_clean.h
  GEN      _security_preproc.h
  GEN      _security.c
  CC       _security_la-_security.lo
  CCLD     _security.la
  COPY     python DSOs
make[4]: Leaving directory '/opt/flux-core/src/bindings/python/_flux'
Making all in flux
make[4]: Entering directory '/opt/flux-core/src/bindings/python/flux'
  GEN      constraint/parsetab.py
```

To fix the warnings in `src/bindings/python/_flux/./make_clean_header.py` you can do each of:

```diff
        with open(f, "r") as header:
            for l in header.readlines():
-                m = re.search('#include\s*"([^"]*)"', l)
+                m = re.search(r'#include\s*"([^"]*)"', l)
                if m:
                    nf = find_first(args.search, m.group(1), including_path)
                    process_header(nf, os.path.dirname(os.path.abspath(nf)))
-                if not re.match("#\s*include", l):
+                if not re.match(r"#\s*include", l):
```
Then in `doc/manpages.py`:

```diff
- ('man3/flux_shell_get_flux', 'flux_shell_get_flux', 'Get a flux_t\* object from flux shell handle', [author], 3),
+ ('man3/flux_shell_get_flux', 'flux_shell_get_flux', r'Get a flux_t\* object from flux shell handle', [author], 3),

...
-    ('man3/flux_shell_service_register', 'flux_shell_service_register', 'Register a service handler for \`method\` in the shell.', [author], 3),
+    ('man3/flux_shell_service_register', 'flux_shell_service_register', r'Register a service handler for \`method\` in the shell.', [author], 3),
-   ('man3/flux_shell_task_channel_subscribe', 'flux_shell_task_channel_subscribe', 'Call \`cb\` when \`name\` is ready for reading.', [author], 3),
+   ('man3/flux_shell_task_channel_subscribe', 'flux_shell_task_channel_subscribe', r'Call \`cb\` when \`name\` is ready for reading.', [author], 3),

```

TLDR: mostly just converting some things to raw strings. You can build with the warnings, but they are ugly looking.

**Comment by grondo** [comment]
Thanks @vsoch. Some other errors were encountered during the conda-forge build with Python 3.12 as well, so I'll get a PR with your suggestions above plus those changes asap.
